clang/include/clang/Basic/AttributeCommonInfo.h

```diff
@@ -94,7 +94,7 @@ class AttributeCommonInfo {
           IsRegularKeywordAttribute(IsRegularKeywordAttribute) {}
     constexpr Form(tok::TokenKind Tok)
         : SyntaxUsed(AS_Keyword), SpellingIndex(SpellingNotCalculated),
-          IsAlignas(Tok == tok::kw_alignas),
+          IsAlignas(Tok == tok::kw_alignas || Tok == tok::kw__Alignas),
```

Comment:
Looking at the activated codepaths, `_Alignas` goes through this (attribute) infrastructure for C parsing as well, but fails to disambiguate the `_Alignas` vs `alignas` information at `isAlignAs` with this change. Ideas in https://github.com/llvm/llvm-project/pull/65638#discussion_r1367333111 requires retaining the distinction rather than mixing with an `||`.

Reply:
If not mixed into `IsAlignAs`, the following can recover `_Alignas`, which [D141177](https://reviews.llvm.org/D141177) is relying on for a lesser invasive change. 

```cpp
getParsedKind() == AT_Aligned && isKeywordAttribute() && !IsAlignas;
```

This feels complicated, and potentially not future proof as in C23, `_Aligned` is an alternate spelling for `aligned`.

Reply:
The other place to recover the `_Alignas` vs `alignas` is `SpellingIndex`. But the following (removed) comment still applies (regarding `SpellingIndex`):

[[clang] Reject non-declaration C++11 attributes on declarations](https://github.com/llvm/llvm-project/commit/8c7b64b5ae2a09027c38db969a04fc9ddd0cd6bb)

```cpp
    // FIXME: Use a better mechanism to determine this
    // We use this in `isCXX11Attribute` below, so it _should_ only return
    // true for the `alignas` spelling, but it currently also returns true
    // for the `_Alignas` spelling, which only exists in C11. Distinguishing
    // between the two is important because they behave differently:
    // - `alignas` may only appear in the attribute-specifier-seq before
    //   the decl-specifier-seq and is therefore associated with the
    //   declaration.
    // - `_Alignas` may appear anywhere within the declaration-specifiers
    //   and is therefore associated with the `DeclSpec`.
    // It's not clear how best to fix this:
    // - We have the necessary information in the form of the `SpellingIndex`,
    //   but we would need to compare against AlignedAttr::Keyword_alignas,
    //   and we can't depend on clang/AST/Attr.h here.
    // - We could test `getAttrName()->getName() == "alignas"`, but this is
    //   inefficient.
```